src/routes/manage_job/post.rs

```diff
@@ -0,0 +1,271 @@
+	type Item = CreateBulkRequestBody;
+	type IntoIter = CreateBulkRequestBodyIterator;
+
+	fn into_iter(self) -> Self::IntoIter {
+		CreateBulkRequestBodyIterator {
+			body: self,
+			index: 0,
+			batch_size: EMAIL_TASK_BATCH_SIZE,
+		}
+	}
+}
+
+impl Iterator for CreateBulkRequestBodyIterator {
+	type Item = CreateBulkRequestBody;
+
+	fn next(&mut self) -> Option<Self::Item> {
+		if self.index <= self.body.input.len() {
+			let item = Some(CreateBulkRequestBody {
+				input_type: self.body.input_type.clone(),
+				input: self.body.input[self.index..self.index + self.batch_size].to_vec(),
+				proxy: self.body.proxy.clone(),
+				hello_name: self.body.hello_name.clone(),
+				from_email: self.body.from_email.clone(),
+				smtp_port: self.body.smtp_port.clone(),
+			});
+
+			self.index = self.index + self.batch_size;
+
+			item
+		} else {
+			None
+		}
+	}
+}
+
+impl Into<CheckEmailInput> for CreateBulkRequestBody {
+	fn into(self) -> CheckEmailInput {
+		let mut input = CheckEmailInput::new(self.input);
+
+		if let Some(name) = self.hello_name {
+			input.set_hello_name(name);
+		}
+
+		if let Some(email) = self.from_email {
+			input.set_from_email(email);
+		}
+
+		if let Some(port) = self.smtp_port {
+			input.set_smtp_port(port as u16);
+		}
+
+		if let Some(proxy) = self.proxy {
+			input.set_proxy(CheckEmailInputProxy {
+				host: proxy.get("host").unwrap().clone(),
+				port: proxy.get("port").unwrap().parse().unwrap(),
+			});
+		}
+
+		input
+	}
+}
+
+/// Endpoint response body.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct CreateBulkResponseBody {
+	job_id: i32,
+}
+
+/// Endpoint error response
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct ErrorResponseBody {
+	error: String,
+}
+
+// Arguments to the `#[job]` attribute allow setting default job options.
+/// This task tries to verify the given email and inserts the results
+/// into the email verification db table
+#[job(channel_name = "foo")]
+pub async fn email_verification_task(
+	mut current_job: CurrentJob,
+	// Additional arguments are optional, but can be used to access context
+	// provided via [`JobRegistry::set_context`].
+) -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+	let task_input: TaskInput = current_job.json()?.unwrap();
+
+	// Retry each future twice, to avoid grey-listing.
+	if let Ok((response, _)) = retry(&task_input.input, RetryOption::Direct, 2).await {
+		let rec = sqlx::query!(
+			r#"
+			INSERT INTO ema_vrfy_rec (job_id, email_id, result)
+			VALUES ($1, $2, $3)
+			"#,
+			task_input.job_id,
+			response.input,
+			serde_json::json!(response)
+		);
+	} else {
+		log::debug!(
+			target:"reacher",
+			"Failed [email={}] for [job={}]",
+			task_input.job_id,
+			task_input.input.to_emails[0]
+		);
+	}
+
+	current_job.complete().await?;
+
+	Ok(())
+}
+
+/// Retry the check ciee_check_email function, in particular to avoid
+/// greylisting.
+#[async_recursion]
+async fn retry(
```

Comment:
Let's re-use this function from the other file?

Reply:
Ok I'll make it `pub` in the other module. However it might be a good idea for later to move the actual verification logic to a separate mod.

Reply:
> a good idea for later to move the actual verification logic to a separate mod.

I agree

Reply:
up

Reply:
Fixed. I've made this `retry` and `CheckEmailError` in the other mod public for now. As we discussed this whole logic can be moved to a separate mod later.